prometheus_metrics/prommetrics.go

```diff
@@ -99,7 +101,40 @@ func (p *PrometheusMetrics) Reset() {
 	p.CronsExecutionTimeHistogram.Reset()
 }
 
+func getAddr(listenAddr string) (string, error) {
+	if listenAddr == "" {
+		return "", fmt.Errorf("Not address provided")
+	}
+
+	// If the address is fine as-is, use it
+	host, port, err1 := net.SplitHostPort(listenAddr)
+	if err1 == nil {
+		return listenAddr, nil
+	}
+
+	// Otherwise, try to add the port
+	listenAddrWithPort := fmt.Sprintf("%s:%d", listenAddr, DefaultPort)
+	host, port, err2 := net.SplitHostPort(listenAddrWithPort)
+	if err2 == nil {
+		return net.JoinHostPort(host, port), nil
```

Comment:
I noticed use of `net.JoinHostPort(host, port)` rather than `listenAddrWithPort` here due to the asymmetry with `return listenAddr` above. Why re-join and why use `fmt.Sprintf` for the first join but `net.JoinHostPort` for the second? Would be interesting to understand your idea.

Reply:
> Why re-join 

Well, the main idea here was that I want to make sure I roundtrip through `SplitHostPort` in order to get some validation (if I don't use it I might end up creating an invalid address and then that's a bit confusing to the user since they'd see an error about using an address they didn't type in). From there I kinda just used `JoinHostPort` without thinking much

> why use fmt.Sprintf for the first join but net.JoinHostPort for the second? Would be interesting to understand your idea.

I guess the above sort of answers this: I used `JoinHostPort` because it kinda fit the mold :) That said, I think it'd be better to use `JoinHostPort` above and not `Sprintf`. The only difference is this would handle `::` as if it were `[::]`, which is convenient.